pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/OfflineSegmentAssignment.java

```diff
@@ -189,45 +163,48 @@ public void init(HelixManager helixManager, TableConfig tableConfig) {
     InstancePartitions instancePartitions = instancePartitionsMap.get(InstancePartitionsType.OFFLINE);
     Preconditions.checkState(instancePartitions != null, "Failed to find OFFLINE instance partitions for table: %s",
         _offlineTableName);
-    LOGGER.info("Rebalancing table: {} with instance partitions: {}", _offlineTableName, instancePartitions);
+    boolean bootstrap =
+        config.getBoolean(RebalanceConfigConstants.BOOTSTRAP, RebalanceConfigConstants.DEFAULT_BOOTSTRAP);
+    LOGGER.info("Rebalancing table: {} with instance partitions: {}, bootstrap: {}", _offlineTableName,
+        instancePartitions, bootstrap);
+    checkReplication(instancePartitions);
 
     Map<String, Map<String, String>> newAssignment;
-    if (instancePartitions.getNumReplicaGroups() == 1) {
-      // Non-replica-group based assignment
+    if (bootstrap) {
+      LOGGER.info("Bootstrapping the table: {}", _offlineTableName);
 
-      List<String> instances =
-          SegmentAssignmentUtils.getInstancesForNonReplicaGroupBasedAssignment(instancePartitions, _replication);
-      newAssignment = SegmentAssignmentUtils
-          .rebalanceTableWithHelixAutoRebalanceStrategy(currentAssignment, instances, _replication);
+      // When bootstrap is enabled, start with an empty table add reassign all segments
```

Comment:
Can you adjust the comment here?

Reply:
What's the suggestion? I feel this comment is clear?